apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts

```diff
@@ -315,63 +315,91 @@ export class ReportFiltersComponent implements OnInit, OnChanges, OnDestroy {
-         this.stockReportRequestExport = {
-             ...this.stockReportRequestExport,
-             showStockName: false,
-             showGroupName: false,
-             showUnitName: false,
-             showOpeningStockQty: false,
-             showOpeningStockValue: false,
-             showInwardsQty: false,
-             showInwardsValue: false,
-             showOutwardsQty: false,
-             showOutwardsValue: false,
-             showClosingStockQty: false,
-             showClosingStockValue: false,
-             showVariantName: false
-         }
-
-        /* for column value filter selected */
-         this.displayedColumns?.forEach(column => {
-             if (column === 'stock_name') {
-                this.stockReportRequestExport.showStockName = true;
-             }
-            else if (column === 'group_name') {
-                this.stockReportRequestExport.showGroupName = true;
-            }
-            else if (column === 'unit_name') {
-                this.stockReportRequestExport.showUnitName = true;
-            }
-            else if (column === 'opening_quantity') {
-                this.stockReportRequestExport.showOpeningStockQty = true;
-            }
-            else if (column === 'opening_amount') {
-                this.stockReportRequestExport.showOpeningStockValue = true;
-            }
-            else if (column === 'inward_quantity') {
+        this.stockReportRequestExport = {
+            ...this.stockReportRequestExport,
+            showStockName: false,
+            showGroupName: false,
+            showUnitName: false,
+            showOpeningStockQty: false,
+            showOpeningStockValue: false,
+            showInwardsQty: false,
+            showInwardsValue: false,
+            showOutwardsQty: false,
+            showOutwardsValue: false,
+            showClosingStockQty: false,
+            showClosingStockValue: false,
+            showVariantName: false,
+            showDate: false,
+            showAccountUniqueName: false,
+            showRate: false,
+            showValue: false
+        }
+
+        /* for column value filter selected common */
+        this.displayedColumns?.forEach(column => {
+            if (column === 'inward_quantity') {
                 this.stockReportRequestExport.showInwardsQty = true;
             }
-            else if (column === 'inward_amount') {
-                this.stockReportRequestExport.showInwardsValue = true;
-            }
             else if (column === 'outward_quantity') {
                 this.stockReportRequestExport.showOutwardsQty = true;
             }
-            else if (column === 'outward_amount') {
-                this.stockReportRequestExport.showOutwardsValue = true;
+            /* for value filter selected in item, variant & group */
+            if (this.searchPage === InventoryReportType.stock || InventoryReportType.variant || InventoryReportType.group) {
```

Comment:
check this once

if (this.searchPage === InventoryReportType.stock || this.searchPage ===  InventoryReportType.variant || this.searchPage ===  InventoryReportType.group) {

aise to nhi likhna chah rhe the ?

Reply:
no
done